torch/utils/data/dataloader.py

```diff
@@ -103,13 +103,15 @@ def default_collate(batch):
         return torch.DoubleTensor(batch)
     elif isinstance(batch[0], string_classes):
         return batch
-    elif isinstance(batch[0], collections.Iterable):
-        # if each batch element is not a tensor, then it should be a tuple
-        # of tensors; in that case we collate each element in the tuple
+    elif isinstance(batch[0], collections.Mapping):
+        return {key: default_collate([d[key] for d in batch]) for key in batch[0]}
+    elif isinstance(batch[0], collections.Sequence):
```

Comment:
I'm not convinced about the change to Sequence. The code is perfectly good for arbitrary Iterables